src/main/java/com/pubnub/api/managers/SubscriptionManager.java

```diff
@@ -273,13 +273,15 @@ private void startSubscribeLoop() {
             @Override
             public void onResponse(SubscribeEnvelope result, PNStatus status) {
                 if (status.isError()) {
-
                     if (status.getCategory() == PNStatusCategory.PNTimeoutCategory) {
                         startSubscribeLoop();
-                    } else {
-                        disconnect();
-                        listenerManager.announce(status);
+                        return;
+                    }
+
+                    disconnect();
+                    listenerManager.announce(status);
 
+                    if (status.getCategory() == PNStatusCategory.PNUnexpectedDisconnectCategory) {
```

Comment:
This seems like it could also be done with an `else if` pattern to make the code a little more readable but this still makes sense.